### PR TITLE
Rewrite ChainerX logic tests

### PR DESCRIPTION
**Pull request by niboshi**
Part of #6401

### CODE REVIEW

File: tests/chainerx_tests/unit_tests/routines_tests/test_logic.py [hunk 1]
```diff
@@ -49,32 +53,41 @@ def _to_array_safe(xp, a_object, dtype):
...
-
-    chainerx.testing.assert_array_equal_ex(
-        cmp_op(a_chx, b_chx), cmp_op(a_np, b_np))
-    chainerx.testing.assert_array_equal_ex(
-        cmp_op(b_chx, a_chx), cmp_op(b_np, a_np))
-    chainerx.testing.assert_array_equal_ex(
-        chx_cmp(a_chx, b_chx), np_cmp(a_np, b_np))
-    chainerx.testing.assert_array_equal_ex(
-        chx_cmp(b_chx, a_chx), np_cmp(b_np, a_np))
+@pytest.mark.filterwarnings('ignore:invalid value encountered in ')
```
Comment (hvy) [inline tests/chainerx_tests/unit_tests/routines_tests/test_logic.py]: Could you leave a comment about in what cases these warnings are raised?

File: tests/chainerx_tests/unit_tests/routines_tests/test_logic.py [hunk 2]
```diff
@@ -104,24 +117,35 @@ def check(x, y):
...
+    ([1],),
+    ([0.2],),
+    ([-0.3],),
+    ([True],),
+    ([False],),
+    ([0, 1, 2],),
+    ([0., numpy.nan],),
+    ([numpy.nan, numpy.inf],),
+    ([-numpy.inf, numpy.nan],),
+    ([[0, 1], [2, 0]],),
```
Comment (hvy) [inline tests/chainerx_tests/unit_tests/routines_tests/test_logic.py]: Is it tricky to align the behavior of `chainer.testing.parameterize_pytest` with `pytest.mark.parametrize`? It's probably outside the scope of this PR so I think it's okay to leave it as is, but for future use-cases.

**Comment by hvy** [comment]
I wrote some minor comments. Could you take a look?

File: tests/chainerx_tests/unit_tests/routines_tests/test_logic.py (hunk 1 above)
Comment (niboshi) [reply tests/chainerx_tests/unit_tests/routines_tests/test_logic.py]: Fixed the comment position.

**Comment by niboshi** [comment]
Rebased

File: tests/chainerx_tests/unit_tests/routines_tests/test_logic.py (hunk 2 above)
Comment (niboshi) [reply tests/chainerx_tests/unit_tests/routines_tests/test_logic.py]: Fixed.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
Travis failures seem unrelated.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**